packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -45,17 +46,44 @@ export type ValidChatModel =
   | 'gpt-3.5-turbo-16k'
   | 'gpt-3.5-turbo-16k-0613';
 
+/**
+ * An OpenAI client that talks to the Azure OpenAI service.
+ * - Adds an API version in the query string.
+ * - Uses Api-Key instead of Authorization for the auth header.
+ * - Includes the deployment name in the path; note that deployment names cannot contain dots.
+ */
+class AzureOpenAIClient extends OpenAIClient {
+  protected override defaultQuery() {
+    return { 'api-version': '2023-07-01-preview' };
+  }
+  protected override authHeaders() {
+    return {
+      'Api-Key': this.apiKey,
+    };
+  }
+  override buildRequest(options: FinalRequestOptions) {
+    if (options.body && 'model' in options.body) {
+      const model = (options.body.model as string).replace('.', '');
+      options.path = `/deployments/${model}${options.path}`;
+    }
+    return super.buildRequest(options);
+  }
+}
+
 const openAiClientContext = AI.createContext<() => OpenAIClient>(
   _.once(() => {
     const baseURL = getEnvVar('OPENAI_API_BASE', false);
-    return new OpenAIClient({
-      apiKey: getEnvVar('OPENAI_API_KEY', false),
+    const useAzure = baseURL?.endsWith('azure.com');
+    const apiKey = getEnvVar(useAzure ? 'OPENAI_AZURE_API_KEY' : 'OPENAI_API_KEY', false);
```

Comment:
Should we fall back to `OPENAI_API_KEY` if `useAzure` and `OPENAI_AZURE_API_KEY` isn't set? Or maybe the OpenAI SDK will already do that?

Reply:
I think we should error in that case (which the OAI SDK will do, I think)